src/pudl/convert/datapkg_to_rst.py

```diff
@@ -0,0 +1,130 @@
+"""
+Module to convert json metadata into rst files.
+
+All of the information about the transformed pudl tables, namely their fields
+types and descriptions, resides in the datapackage metadata. This module makes
+that information available to users, without duplicating any data, by converting
+json metadata files into documentation-compatible rst files. The functions
+serve to extract the field names, field data types, and field descriptions of
+each pudl table and outputs them in a manner that automatically updates the
+read-the-docs.
+"""
+
+import argparse
+import json
+import logging
+import sys
+
+from jinja2 import Template
+
+logger = logging.getLogger(__name__)
+
+
+###############################################################################
+# T E M P L A T E S
+###############################################################################
+
+"""
+The following templates map json data into one long rst file seperated by table
+titles and document links (RST_TEMPLATE)
+
+It's important for the templates that the json data do not contain excess
+white space either at the beginning or the end of each value.
+"""
+
+
+# Template for all tables in one rst file
+RST_TEMPLATE = '''
+===============================================================================
+All PUDL Database Tables
+===============================================================================
+
+{% for resource in resources %}
+.. _{{ resource.name }}:
+
+-------------------------------------------------------------------------------
+{{ resource.name }}
+-------------------------------------------------------------------------------
+
+.. list-table::
+  :widths: auto
+  :header-rows: 1
+
+  * - **Field Name**
+    - **Type**
+    - **Description**
+{% for field in resource.schema.fields %}
+  * - {{ field.name }}
+    - {{ field.type }}{% if field.description %}
+    - {{ field.description }}{% else %}
+    - N/A{% endif %}
+{% endfor %}
+{% endfor %}
+'''
+""" A template to map data from a json dictionary into one rst file. Contains
+    multiple tables seperated by headers.
+"""
+
+###############################################################################
+# F U N C T I O N S
+###############################################################################
+
+
+def datapkg2rst_one(meta_json, meta_rst):
+    """Convert json metadata to a single rst file."""
+    logger.info("Accessing json metadata as dictionary")
+    with open(meta_json) as f:
+        metadata_dict = json.load(f)
+
+    metadata_dict["resources"] = sorted(
+        metadata_dict["resources"],
+        key=lambda x: x["name"]
+    )
```

Comment:
This sorts the list of tables so that it's a little easier to find one if you're looking for it in the list by name.